modules/eks-managed-node-group/variables.tf

```
@@ -0,0 +1,461 @@
+variable "create" {
+  description = "Determines whether to create EKS managed node group or not"
+  type        = bool
+  default     = true
+}
+
+variable "tags" {
+  description = "A map of tags to add to all resources"
+  type        = map(string)
+  default     = {}
+}
+
+variable "platform" {
+  description = "Identifies if the OS platform is `bottlerocket` or `linux` based; `windows` is not supported"
+  type        = string
+  default     = "linux"
+}
+
+################################################################################
+# User Data
+################################################################################
+
+variable "enable_bootstrap_user_data" {
+  description = "Determines whether the bootstrap configurations are populated within the user data template"
+  type        = bool
+  default     = false
+}
+
+variable "cluster_name" {
+  description = "Name of associated EKS cluster"
+  type        = string
+  default     = null
+}
+
+variable "cluster_endpoint" {
+  description = "Endpoint of associated EKS cluster"
+  type        = string
+  default     = ""
+}
+
+variable "cluster_auth_base64" {
+  description = "Base64 encoded CA of associated EKS cluster"
+  type        = string
+  default     = ""
+}
+
+variable "pre_bootstrap_user_data" {
+  description = "User data that is injected into the user data script ahead of the EKS bootstrap script. Not used when `platform` = `bottlerocket`"
+  type        = string
+  default     = ""
+}
+
+variable "post_bootstrap_user_data" {
+  description = "User data that is appended to the user data script after of the EKS bootstrap script. Not used when `platform` = `bottlerocket`"
+  type        = string
+  default     = ""
+}
+
+variable "bootstrap_extra_args" {
+  description = "Additional arguments passed to the bootstrap script. When `platform` = `bottlerocket`; these are additional [settings](https://github.com/bottlerocket-os/bottlerocket#settings) that are provided to the Bottlerocket user data"
+  type        = string
+  default     = ""
+}
+
+variable "user_data_template_path" {
+  description = "Path to a local, custom user data template file to use when rendering user data"
+  type        = string
+  default     = ""
+}
+
+################################################################################
+# Launch template
+################################################################################
+
+variable "create_launch_template" {
+  description = "Determines whether to create a launch template or not. By default, EKS will use its own default launch template"
+  type        = bool
+  default     = false
```

```suggestion
  description = "Determines whether to create a launch template or not. If set to `false`, EKS will use its own default launch template"
  type        = bool
  default     = true
```

There are no downsides to creating a launch template when using this module and many features require enabling launch templates. Enabling this by default will simplify use of this module greatly.

---

Agreed - can't believe I forgot to flip this. It actually solves the whole conundrum of tags on ASGs without any hacky efforts 
- #1558
- #1675
- #1688
- #1689
- #1705
- #1711